Patch-Source: https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/merge_requests/20

From 6c6a42e37160fbe7d6d44960f664f8b4c85c587d Mon Sep 17 00:00:00 2001
From: knuxify <knuxify@gmail.com>
Date: Tue, 21 Dec 2021 08:19:39 +0100
Subject: [PATCH] meson.build: add option to disable systemd service file build

On systems without systemd the service file isn't necessary.
This commit adds automatic systemd detection (that can be
overridden with the -Dsystemd_service option), which determines
whether the service file is built and installed.
---
 data/meson.build  | 14 ++++++++------
 meson.build       | 25 +++++++++++++++++--------
 meson_options.txt |  5 +++++
 3 files changed, 30 insertions(+), 14 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index e62ed10..4b0b1e6 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -16,12 +16,14 @@ configure_file(
 )
 
 # systemd unit
-configure_file(
-  input: 'xdg-desktop-portal-gnome.service.in',
-  output: 'xdg-desktop-portal-gnome.service',
-  configuration: libexecdir_conf,
-  install_dir: systemduserunitdir,
-)
+if systemduserunitdir != ''
+  configure_file(
+    input: 'xdg-desktop-portal-gnome.service.in',
+    output: 'xdg-desktop-portal-gnome.service',
+    configuration: libexecdir_conf,
+    install_dir: systemduserunitdir,
+  )
+endif
 
 # Desktop file
 desktop_in = configure_file(
diff --git a/meson.build b/meson.build
index a112cf1..9b45b82 100644
--- a/meson.build
+++ b/meson.build
@@ -10,14 +10,18 @@ prefix = get_option('prefix')
 datadir = get_option('datadir')
 libdir = get_option('libdir')
 libexecdir = get_option('libexecdir')
+build_systemd_service = get_option('systemd_service')
 
-systemduserunitdir = get_option('systemduserunitdir')
-if systemduserunitdir == ''
-  systemd = dependency('systemd', version: '>= 242')
-  systemduserunitdir = systemd.get_pkgconfig_variable(
-    'systemduserunitdir',
-    define_variable: ['prefix', get_option('prefix')]
-  )
+systemduserunitdir = ''
+systemd = dependency('systemd', version: '>= 242', required: build_systemd_service)
+if systemd.found() and not build_systemd_service.disabled()
+  systemduserunitdir = get_option('systemduserunitdir')
+  if systemduserunitdir == ''
+    systemduserunitdir = systemd.get_pkgconfig_variable(
+      'systemduserunitdir',
+      define_variable: ['prefix', get_option('prefix')]
+    )
+  endif
 endif
 
 dbus_service_dir = get_option('dbus_service_dir')
@@ -35,5 +39,10 @@ summary({
   'libdir': libdir,
   'libexecdir': libexecdir,
   'dbus_service_dir': dbus_service_dir,
-  'systemduserunitdir': systemduserunitdir,
+  'systemd_service': build_systemd_service.enabled() or build_systemd_service.auto(),
 })
+if build_systemd_service.enabled() or build_systemd_service.auto()
+  summary({
+    'systemduserunitdir': systemduserunitdir,
+  })
+endif
diff --git a/meson_options.txt b/meson_options.txt
index 7953401..f45a62f 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -3,6 +3,11 @@ option('dbus_service_dir',
   description: 'Directory for D-Bus service files'
 )
 
+option('systemd_service',
+  type: 'feature',
+  description: 'Whether or not the systemd user service should be built'
+)
+
 option('systemduserunitdir',
   type: 'string',
   description: 'Directory for systemd user service files'
-- 
GitLab

